### PR TITLE
#112: Add documentation

### LINKED ISSUE #112: Deploy documentation via Doxygen
@lifflander if you want to provide more details here

The documentation repository is DARMA-tasking/DARMA-tasking.github.io repository.

### PR DESCRIPTION
**Pull request by maxime-bfsquall**
Fixes #112

### CODE REVIEW

**Comment by maxime-bfsquall** [comment]
Waiting for a way to "commit" and "push" the generated documentation using another method.

**Current method based on VT** : Defining a user (name/email) and using a token passed by the variable secrets.GH_PAT to the CI

**Comment by maxime-bfsquall** [comment]
Before the merge, we have to change the documentation GIT branch used for tests `112-add-documentation` to `master`

File: docs/md/mainpage.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+\section ckpt_what_is What is VT TV?
+VT TV is ...
+
+\section License
+
+@m_class{m-note m-dim}
+
+@parblock
+Copyright 2021-2024 National Technology & Engineering Solutions of Sandia, LLC
```
Comment (cwschilly) [inline docs/md/mainpage.md:13]: @maxime-bfsquall @ppebay @lifflander The LICENSE file at the top of the repo says `Copright 2019`. Which one is correct?

**Comment by cwschilly** [comment]
I added the README from PR #59 into `mainpage.md` -- once that PR is approved I'll update this one with any more small changes

File: docs/md/mainpage.md (hunk 1 above)
Comment (ppebay) [reply docs/md/mainpage.md:300]: We should make sure that this verification is made common to repositories @maxime-bfsquall @cwschilly

**Review by ppebay — APPROVED** [review]

**Comment by tlamonthezie** [comment]
> Before the merge, we have to change the documentation GIT branch used for tests `112-add-documentation` to `master`

Hello @maxime-bfsquall I think you can do these changes now is has been already been approved

**Pull request merged**